runners/direct-java/src/main/java/org/apache/beam/runners/direct/portable/RootProviderRegistry.java

```diff
@@ -33,6 +34,16 @@
  * based on the type of {@link PTransform} of the application.
  */
 class RootProviderRegistry {
+  public static RootProviderRegistry javaPortableRegistry(BundleFactory bundleFactory) {
+    return new RootProviderRegistry(
+        ImmutableMap.<String, RootInputProvider<?>>builder()
+            .put(
+                IMPULSE_TRANSFORM_URN,
+                new ImpulseEvaluatorFactory.ImpulseRootProvider(bundleFactory))
+            .put(FLATTEN_TRANSFORM_URN, new EmptyInputProvider())
+        .build());
+  }
+
   /** Returns a {@link RootProviderRegistry} that only supports the {@link Impulse} primitive. */
   public static RootProviderRegistry impulseRegistry(BundleFactory bundleFactory) {
```

Comment:
Shall we remove impulse registry as it is not used anymore?

Reply:
Agreed. Done.